test/cql-pytest/test_native_transport.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+#############################################################################
+# The cql-pytest framework is about testing CQL functionality, so
+# implementation details like sstables cannot be tested directly. However,
+# we are still able to reproduce some bugs by tricks such as writing some
+# data to the table and then force it to be written to the disk (nodetool
+# flush) and then trying to read the data again, knowing it must come from
+# the disk.
+#############################################################################
+
+import pytest
+import time
+import nodetool
+from util import new_test_table, new_cql
+from cassandra.cluster import NoHostAvailable
+
+def test_enable_disable_binary(cql, test_keyspace):
+    schema = 'k text, v text, primary key (k)'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"INSERT INTO {table} (k,v) VALUES ('foo', 'bar')")
+        cql.execute(f"SELECT * FROM {table}")
+
+        nodetool.disablebinary(cql)
+        select_success = False
+        try:
+            cql.execute(f"SELECT * FROM {table}")
+            select_success = True
+        except NoHostAvailable:
+            pass
+        assert not select_success
```

Comment:
pytest allows you to replace these 7 lines with the much shorter and clearer:
```
with pytest.raises(NoHostAvailable):
    cql.execute(f"SELECT * FROM {table}")
```

I know the loop below is a bit more complicated, but this one isn't.